baseplate/context/thrift.py

```diff
@@ -48,6 +48,13 @@ def __init__(self, pool, client_cls):
             },
         )
 
+    def report_runtime_metrics(self, batch):
+        batch.gauge("size").replace(self.pool.size)
+        batch.gauge("checkedout").replace(self.pool.checkedout)
+        # it's hard to report "checkedin" currently because we can't
+        # distinguish easily between available connection slots that aren't
+        # instantiated and ones that have actual open connections.
```

Comment:
Yeah! I'm totally down for a different set of names. I definitely just copied sqlalchemy's for the thrift pool. Got any thoughts on what would be clearest? Do we have any prior art elsewhere?

Reply:
`size`: A+, would use again
`overflow`: I'm not 100% sure what this is but I would guess it's requests that are waiting for an available slot in the pool?  `overflow` seems appropriate or `waiting` could also work.
`checkedin`: `available`, `open`.
`checkedout`: `in_use`/`inuse`/`in-use`, `active`, `used`

I don't think that any of the sqlalchemy names are crazy or super specific to the library so I think they work too.

Reply:
Note, I'm not _aware_ of any prior art.

Reply:
🔕 Sort of related, should these be `pool.${metric}`?  That way we can potentially add other metrics that aren't explicitly related to the connection pools and have those name-spaced appropriately?  Or is that overkill?

Reply:
`overflow` is a SQLAlchemy pool thing. It's a number of connections that can be opened beyond the max size of the pool, but if they're returned to the pool they're closed rather than kept around. So they allow spike capacity but don't stick around for the long haul.

I'm cool with `available` and `in_use`. That sounds clear to me.

Yeah I think `pool.` prefix is fine since it'd allow easy wildcards.

Reply:
I like it all!

Reply:
I had to change to `open_and_available` because it's different from `size - in_use` in that it's explicitly telling us how many TCP connections are open but checked back into the pool; it does not count how many connection slots are free.